File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/section_4_2_downward_dependency/after/SqlOrderRepository.java
package com.grokkingsoftwarearchitecture.chapter04.section_4_2_downward_dependency.after;


/**
 * Insfrastructure Layer
 * * ARCHITECTURE NOTE:
 * This concrete implementation sits in the Infrastructure layer 
 * and implements the interface defined by the domain.
 */
public class SqlOrderRepository implements OrderRepository {
    @Override
    public void save(Order order) {
        System.out.println("(After Refactor) Saving order to SQL...");
    }
}

File: Chapter 04/C#/Section_4_2_Downward_Dependency/After/SQLOrderRepository.cs
namespace Chapter04.Section_4_2.After
{
    // DataAccessLayer.cs  
	public class SqlOrderRepository : IOrderRepository // Implements interface 
	{
	    public void Save(Order order)
	    {
	        Console.WriteLine("(After Refactor) Saving order to SQL...");
	    }
	}
}